src/pgrn-custom-scan.c

```diff
@@ -351,22 +364,55 @@ PGrnExecCustomScan(CustomScanState *customScanState)
 
 	{
 		TupleTableSlot *slot = customScanState->ss.ps.ps_ResultTupleSlot;
+		unsigned int i = 0;
+		ListCell *cell;
 
 		ExecClearTuple(slot);
-		for (unsigned int i = 0; i < slot->tts_tupleDescriptor->natts; i++)
+		// We might add state->targetlist instead of ss.ps.plan->targetlist.
+		foreach (cell, customScanState->ss.ps.plan->targetlist)
 		{
-			Form_pg_attribute attr =
-				TupleDescAttr(slot->tts_tupleDescriptor, i);
-			grn_obj *column = GRN_PTR_VALUE_AT(&(state->columns), i);
+			TargetEntry *entry = (TargetEntry *) lfirst(cell);
 			GRN_BULK_REWIND(&(state->columnValue));
-			grn_obj_get_value(ctx, column, id, &(state->columnValue));
-			slot->tts_values[i] =
-				PGrnConvertToDatum(&(state->columnValue), attr->atttypid);
-			// todo
-			// If there are nullable columns, do not custom scan.
-			// See also
-			// https://github.com/pgroonga/pgroonga/pull/742#discussion_r2107937927
-			slot->tts_isnull[i] = false;
+
+			if (IsA(entry->expr, Var))
+			{
+				Oid typeId = exprType((Node *) entry->expr);
+				grn_obj *column = GRN_PTR_VALUE_AT(&(state->columns), i);
+				grn_obj_get_value(ctx, column, id, &(state->columnValue));
+				slot->tts_values[i] =
+					PGrnConvertToDatum(&(state->columnValue), typeId);
+				// todo
+				// If there are nullable columns, do not custom scan.
+				// See also
+				// https://github.com/pgroonga/pgroonga/pull/742#discussion_r2107937927
+				slot->tts_isnull[i] = false;
+			}
+			else if (IsA(entry->expr, FuncExpr))
+			{
+				FuncExpr *funcExpr = (FuncExpr *) entry->expr;
+				if (strcmp(get_func_name(funcExpr->funcid), "pgroonga_score") ==
+					0)
+				{
+					grn_obj_get_value(
```

Comment:
TODO: Reject this function if the argument isn't `(tableoid, ctid)` nor `(record)`.